### ISSUE TITLE
Why is the mAP so different from the official mAP when I train YOLOv8s from scratch?

### ISSUE THREAD

**Issue by momentNi** [ultralytics/ultralytics#19241]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Recently, I tried to modify the network structure of YOLOv8. I modified the structure and train it under the direction of official docs with full COCO dataset (coco.yaml in official code). However, the mAP in results.csv is very low. **Even if I tried to train the original YOLOv8s structure, I still got about 0.28 mAP50-95 and 0.45 mAP50, which is very low compared to 0.449 mAP50-95 and 0.61 mAP50 in official docs**. The train process even early stopped at 69th epoch when i set patience to 10.

I see a big improvement at the begining of the the normal train process while in my situation it can't be seen. So, this situation really confused me. I NEED URGENT HELP!!!

Here's my code while training original YOLOv8s from scratch:

```python
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO('yolov8s.yaml')
    model.train(
        data=r"coco.yaml",
        cache=False,
        imgsz=640,
        epochs=150,
        patience=10,
        single_cls=False,
        batch=64,
        optimizer='SGD',
        seed=42,
        amp=True,
        project='runs/train',
        name='yolov8s',
    )
```

Here's my results.csv when training original YOLOv8s:

![Image](https://github.com/user-attachments/assets/af1fe15f-dab8-48d8-ba59-e7326e24f7b0)

![Image](https://github.com/user-attachments/assets/70fac2c6-f5aa-4742-8363-b67a6d1f89ca)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @momentNi, thank you for your interest in Ultralytics 🚀! We appreciate you sharing your results and efforts. Let's work together on this!

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) including the exact code, dataset, and steps required to reproduce the issue you’re encountering. This is crucial to help us pinpoint the root cause effectively.

If this is a custom training ❓ Question:
1. Please ensure you are following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
2. Share additional details if possible, such as dataset samples, complete training logs, and any modifications you made to the network or training setup.

To debug further:
- Ensure you have the latest `ultralytics` package and all dependencies installed. Update using:

```bash
pip install -U ultralytics
```

- Confirm your development environment is set up correctly. YOLO supports the following verified environments with all dependencies ([Python](https://www.python.org/) >=3.8, [PyTorch](https://pytorch.org/) >=1.8, [CUDA](https://developer.nvidia.com/cuda-zone)) preinstalled:
  - **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
  - **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
  - **Google Cloud** or **AWS Deep Learning** VMs. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).

For structured experiments, ensure your setup is aligned with the official training settings provided in our [Docs](https://docs.ultralytics.com/) or [Tasks](https://docs.ultralytics.com/tasks/) sections.

Join the Ultralytics community for additional support if needed! 
- Chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- Share insights on [Discourse](https://community.ultralytics.com/) or the [Subreddit](https://reddit.com/r/Ultralytics)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, validating [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on multiple platforms daily.

PS: This is an automated response. An Ultralytics engineer will also assist you soon. Let us know if we can help clarify anything further in the meantime! 😊

**Comment by Y-T-G** [comment]
You can find the training arguments used from the model.

https://github.com/ultralytics/ultralytics/issues/17048#issuecomment-2424742030

**Comment by Y-T-G** [comment]
Also what version of Ultralytics is this

**Comment by momentNi** [comment]
> You can find the training arguments used from the model.
> 
> [#17048 (comment)](https://github.com/ultralytics/ultralytics/issues/17048#issuecomment-2424742030)

Thanks for your help! I will try it now! My version of Ultralytics is 8.3.59.

So the small improvement at the begining of my train is probably caused by my inappropriate hyper args?

**Comment by glenn-jocher** [comment]
Yes, the lower mAP and slow initial improvement likely stem from suboptimal hyperparameters, and reviewing/adjusting them as per the linked comment may help, along with ensuring your dataset configuration in `coco.yaml` is correct.
